### PR TITLE
Allow player to offer control of their mob to ghosts.

### PR DESCRIPTION
**Pull request by Fhaxaris**
## About The Pull Request
Allows players to offer control of their own mob using the ghost verb. 

## Why It's Good For The Game
Less catatonic people from ghosting. (people don't admin help when they ghost as a minor role)

## Changelog
:cl:
add: You can now use the ghost verb to offer control of your mob.
/:cl:

### CODE REVIEW

**Comment by wesoda25** [comment]
Make sure its not spammable if you haven't already

**Comment by SumFaggotPlayTester** [comment]
This is good. Very useful.

**Comment by SkeletalElite** [comment]
Perhaps make it so if no one responds you go catatonic but your mob becomes posses-sable in case someone else wants to take it later?

**Comment by Ryll-Ryll** [comment]
There are already IC ways of handling people leaving their positions by just going to the HoP and asking to be their job. Offering players to ghosts is a bad solution that should only be taken when an important role like an antag or a head of staff has to leave immediately, because of the administrative issues it creates (example: player A kills player B, player A leaves and is replaced by player C, player B gets revenge on player A/C, player C ahelps because it appears random to them)

Honestly as an admin I do as much as I can to tell people to just promote one of their subordinates when they have to leave rather than offering them to ghosts, and if a minor role like the curator or the chef ahelps, I just tell them they're not important enough to offer and to just enjoy their sleep/dinner/whatever. Being able to offer yourself to ghosts is an OOC solution to a problem that can and should easily be handled ICly

**Comment by SkeletalElite** [comment]
> Honestly as an admin I do as much as I can to tell people to just promote one of their subordinates when they have to leave rather than offering them to ghosts, and if a minor role like the curator or the chef ahelps, I just tell them they're not important enough to offer and to just enjoy their sleep/dinner/whatever. Being able to offer yourself to ghosts is an OOC solution to a problem that can and should easily be handled ICly

I think its mainly for like head of staff and ghost roles like the comms agent/syndi scientist

**Comment by Fhaxaris** [comment]
>  if a minor role like the curator or the chef ahelps, I just tell them they're not important enough to offer and to just enjoy their sleep/dinner/whatever.

This is the issue the PR is addressing. No admin or player wants to go through the trouble of ahelping when they are a minor role and have to leave. This means players just ghost when they are not a head role and need to leave. The player doesn't have to ahelp, and the button to offer control is right next to the button to confirm the ghost.

> Being able to offer yourself to ghosts is an OOC solution to a problem that can and should easily be handled ICly

Something which is an OOC problem (them having to leave) typically should not be solved ICly. When you want to die, you don't beat yourself to death or ask someone to kill you, you use the suicide verb because randomly wanting to die almost always an OOC decision, not an IC one.

**Comment by Ryll-Ryll** [comment]
Again, heads of staff should be promoting one of their subordinates ICly (even just saying "first one to my office gets my job" and ghosting in there). It'd be fine if it was just for ghost roles, but there isn't a limit on this PR for that

**Comment by ATH1909** [comment]
> Again, heads of staff should be promoting one of their subordinates ICly (even just saying "first one to my office gets my job" and ghosting in there). It'd be fine if it was just for ghost roles, but there isn't a limit on this PR for that
> 
> And having to leave is as much an IC issue as it is an OOC issue. People have to leave suddenly in real life, tell them you have an appointment elsewhere before you leave then hide yourself out of sight, you don't need much of an IC excuse

I've been straight up banned for suiciding as a head of staff AFTER tossing my ID to a scientist, because I didn't ask for permission from an admin to do so first.

How is that different from suiciding as a head of staff after promoting someone else without asking an admin first?

**Comment by Fhaxaris** [comment]
> Again, heads of staff...

Dude, this isn't about heads, it's about lesser roles. When one is a head role, they are more accountable and already required to take action before they leave, there is no issue there. When one is a chef, janitor, medical doctor, engineer, or security officer, they are allowed to ghost or suicide without penalty despite the fact that those roles are also important to round progression. Letting players offer control will let them leave while having less of an effect on the round. They are already going to ghost because nothing is stopping them, they might as well offer control first.

You keep responding with alternative actions players can take, but there is no rule saying they must do these things(as a non-head role), so there is no incentive to make the effort to improve the round other than their own goodwill.

Also, you're talking as if this PR will make your alternative suggestions impossible. It's just giving more options.

**Comment by Arianya** [comment]
To be clear, shoving ghosts into a head/antagonist is a workaround to an issue with round flow - I don't think it's well suited to mass use like this.

To be clear, what issue does this seek to solve? The PR opener says 
> Less catatonic people from ghosting. (people don't admin help when they ghost as a minor role)
But I don't see people being catatonic because they're choosing to ghost/disconnect as being an issue in and of itself. I would consider ghosts with potential salt or metaknowledge using this as a very easy means back on station as a much bigger issue.

**Comment by ExcessiveUseOfCobblestone** [comment]
@ATH1909 if this was on tg you do not need to ask for permission to leave, you just need to notify an admin that you are leaving since you are labeled as a high-importance role (preferably explaining youre giving up your role to someone else ingame). I'm even fine with F1 > "g2g"

> You keep responding with alternative actions players can take, but there is no rule saying they must do these things(as a non-head role), so there is no incentive to make the effort to improve the round other than their own goodwill.

You act like there is a req to play the role you get a second life in. If I die as a doctor and get an option for chef, what would be stopping me from going to the hop to get my MD job back? Add that to the fat I can """find""" my old gear and """find""" hints that reveal who killed me. You could argue ghost roles can do it too, and I'd agree if we didn't try to keep them away from the station for that very reason.

Just looking at this from a gameplay perspective I imagine it wont be very fun having to hope that the guy you killed isn't going to use the newfound assistant catatonic to "stumble upon" his body. It will be a real pain to enforce because I will now have to logdive to connect the dots.

File: code/modules/mob/dead/observer/observer.dm [hunk 1]
```diff
@@ -290,7 +290,9 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 	if(stat == DEAD)
 		ghostize(TRUE)
 		return TRUE
-	var/response = alert(src, "Are you -sure- you want to ghost?\n(You are alive. If you ghost whilst still alive you may not play again this round! You can't change your mind so choose wisely!!)","Are you sure you want to ghost?","Ghost","Stay in body")
+	var/response = alert(src, "Are you -sure- you want to ghost?\n(You are alive. If you ghost or offer another player control of your character whilst still alive you may not play again this round! You can't change your mind so choose wisely!!)","Are you sure you want to ghost?","Ghost","Offer control","Stay in body")
+	if(response == "Offer control")
+		return offer_control(usr)
 	if(response != "Ghost")
 		return FALSE//didn't want to ghost after-all
 	ghostize(FALSE)						// FALSE parameter is so we can never re-enter our body. U ded.
```
Comment (nemvar) [inline code/modules/mob/dead/observer/observer.dm:298]: :smirk:
This will ghostize you if you click "Stay in body"

**Comment by wesoda25** [comment]
Why not just enable this only if there are no admins online, and only for important jobs or roles (heads/antags)?

**Comment by BeeSting12** [comment]
I think this will create more administrative issues than it will solve as Cobby, Shaps, and Arianya said.

**Comment by HomocidlePancakes** [comment]
Just restrict it to antags. Problem solved.

**Pull request closed without merging**